cli.go

```diff
@@ -198,6 +190,68 @@ func (i *CLI) Run() {
 	}
 }
 
+func (i *CLI) setupHeartbeat() {
+	hbURL := i.c.String("heartbeat-url")
+	if hbURL == "" {
+		return
+	}
+
+	hbTok := i.c.String("heartbeat-url-auth-token")
+	if strings.HasPrefix(hbTok, "file://") {
+		hbTokBytes, err := ioutil.ReadFile(strings.Split(hbTok, "://")[1])
+		if err != nil {
+			i.logger.WithField("err", err).Error("failed to read auth token from file")
+		} else {
+			hbTok = string(hbTokBytes)
+		}
+	}
+
+	i.logger.WithField("heartbeat_url", hbURL).Info("starting heartbeat loop")
+	go i.heartbeatHandler(hbURL, strings.TrimSpace(hbTok))
+}
+
+func (i *CLI) handleStartHook() {
+	hookValue := i.c.String("start-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("start_hook", hookValue).Info("running")
```

Comment:
Would a better message be "running hook" or "running start hook"? I'm thinking if we want to search for these log messages, searching for `msg="running start hook"` might be nicer than `msg="running" start_hook=`?